lib/autobahn/consumer.rb

```diff
@@ -59,6 +59,13 @@ def unsubscribe!(timeout=30)
             subscription.cancel
           end
           @subscriptions = nil
+          @deliver.set(false)
```

Comment:
You're correct, with this change we might process a portion of a batch and then just call it a day, which wouldn't be a good thing at all. I've changed how this works in the latest version.